test/server/cards/01_SOR/upgrades/SmugglingCompartment.spec.ts

```diff
@@ -0,0 +1,51 @@
+describe('Smuggling Compartment', function() {
+    integration(function(contextRef) {
+        describe('Smuggling Compartment\'s on attack ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [{ card: 'atst', upgrades: ['smuggling-compartment'] }, { card: 'wampa', upgrades: ['smuggling-compartment'] }],
+                    },
+                    player2: {
+                        groundArena: ['snowspeeder']
+                    }
+                });
+            });
+
+            it('should ready a resource on attack', function () {
+                const { context } = contextRef;
+
+                context.player1.exhaustResources(2);
+
+                const exhaustedResourcesBeforeAbility = context.player1.countExhaustedResources();
+
+                context.player1.clickCard(context.atst);
+                context.player1.clickCard(context.snowspeeder);
+
+                expect(context.player1.countExhaustedResources()).toBe(exhaustedResourcesBeforeAbility - 1);
+            });
+        });
+
+        describe('Smuggling Compartment', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['smuggling-compartment'],
+                        groundArena: ['snowspeeder', 'battlefield-marine']
```

Comment:
maybe add some units on space arena and on player 2 to check arena & controller ?

Reply:
I'm never opposed to more tests but I think it's fine without that, we have tests for checking general attachment rules like that in Upgrade.spec.ts so there's no real need to have those tests included in suites for individual upgrades